bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -70,12 +74,28 @@ def apply_prep(target_spec)
 
           unless need_install.empty?
             need_install_targets = need_install.map(&:target)
-            run_task(executor, need_install_targets, 'puppet_agent::install')
-            # Service task works best when targets have puppet-agent feature
+
+            plugin_targets, builtin_targets = need_install_targets.partition do |t|
```

Comment:
We should just use the the `InstallAgent` Plugin as a default in config rather then treating these nodes specially.